libqtile/widget/base.py

```diff
@@ -510,7 +510,16 @@ def worker():
             try:
                 text = self.poll()
                 if self.qtile is not None:
+                    # This can throw a runtime error if the event loop is
+                    # not available. This would only happen before qtile is
+                    # started or after it is finalized. The thread here is
+                    # fire and forget, so cannot be signaled.
                     self.qtile.call_soon_threadsafe(self.update, text)
+            except RuntimeError:  # gets raised when the event loop is down
+                logger.debug(
+                    'Thread tried to access event loop while it was unavailable',
```

Comment:
We don't know this for sure, a RuntimeError could happen for other reasons, and this is why I'd prefer a debug log in the first place. I'd rather write something like: `problem polling to update widget %s (ignoring, this is most likely because the event loop has finalized already)`. And then you could probably remove the other comments. :)

Reply:
Or we can match against the error message and re-raise if it's different than what we expect.

Reply:
Agreed; IMO a stack trace is useful to track these sorts of things down, so let's include that if we can.